### PR TITLE
python implementation

### PR DESCRIPTION
**Pull request by knhngchn**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: content/4_Gold/MST.mdx [hunk 1]
```diff
@@ -495,6 +495,46 @@ class Pair implements Comparable<Pair> {
 
 </JavaSection>
 
+<PySection>
+
+```py
+import sys
+import heapq
+input = lambda: sys.stdin.readline().rstrip("\r\n")
```
Comment (SansPapyrus683) [inline content/4_Gold/MST.mdx:503]: please don't do this

File: content/4_Gold/MST.mdx [hunk 2]
```diff
@@ -495,6 +495,46 @@ class Pair implements Comparable<Pair> {
...
+	a, b, c = map(int, input().split())
+	a -= 1; b -= 1
+	G[a].append((c, b))
+	G[b].append((c, a))
+
+ans = prim()
+if ans == -1:
+	print("IMPOSSIBLE")
+else:
+	print(ans)
```
Comment (SansPapyrus683) [inline content/4_Gold/MST.mdx:533]:
```suggestion
print('IMPOSSIBLE' if ans == -1 else ans)
```

File: content/4_Gold/MST.mdx [hunk 3]
```diff
@@ -495,6 +495,46 @@ class Pair implements Comparable<Pair> {
...
+		count += 1
+		for (neigh_weight, neigh_node) in G[node]:
+			heapq.heappush(pq, (neigh_weight, neigh_node))
+	return total_weight if count == n else -1
+
+n, m = map(int, input().split())
+G = [[] for _ in range(n)]
+for _ in range(m):
+	a, b, c = map(int, input().split())
+	a -= 1; b -= 1
```
Comment (SansPapyrus683) [inline content/4_Gold/MST.mdx:525]: statements on one line

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
**also did you make sure it AC's? send screenie if it did**

File: content/4_Gold/MST.mdx (hunk 2 above)
Comment (knhngchn) [reply content/4_Gold/MST.mdx]: Updated.

File: content/4_Gold/MST.mdx (hunk 3 above)
Comment (knhngchn) [reply content/4_Gold/MST.mdx]: Updated.

**Comment by knhngchn** [comment]
@SansPapyrus683
Thanks for reviewing.
I just updated some of the code, and here's the screenshot:
https://imgur.com/BHQNJG1

File: content/4_Gold/MST.mdx (hunk 1 above)
Comment (knhngchn) [reply content/4_Gold/MST.mdx]: Removed.

**Review by SansPapyrus683 — APPROVED** [review]
don't like the global variables, but other than that it's fine

**Comment by SansPapyrus683** [comment]
no i mean like pass the argument (graph, # of nodes, etc.) in the function as arguments

**Comment by knhngchn** [comment]
@SansPapyrus683
Ok I just fixed it.

File: content/4_Gold/MST.mdx [hunk 4]
```diff
@@ -495,6 +495,47 @@ class Pair implements Comparable<Pair> {
...
+		a -= 1
+		b -= 1
+		G[a].append((c, b))
+		G[b].append((c, a))
+
+	ans = prim(n, G)
+	print("IMPOSSIBLE" if ans == -1 else ans)
+
+if __name__=='__main__':
+	main()
```
Comment (SansPapyrus683) [inline content/4_Gold/MST.mdx:534]: you can take this out of main

Comment (knhngchn) [reply content/4_Gold/MST.mdx]: Updated.

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**